conformance/lang/expressions/string-template-expr/string_template_expr.balt

```diff
@@ -0,0 +1,323 @@
+Test-Case: output
+Description: Test whether string-template-expr interpolates the results of evaluating expressions into a literal string.
+Labels: string, function-defn, int, string-template-expr, string-literal
+
+function init() {
+    string a1 = "a1";
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output a1
+    int a2 = 12;
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output 12
+    int a3 = 13;
+    string temp3 = string `${a2 + a3}`;
+    io:println(temp3); // @output 25
+    string temp4 = string `${"a2"}`;
+    io:println(temp4); // @output a2
+}
+
+Test-Case: error
+Description: Test whether string-template-expr interpolates the results of evaluating expressions into a literal string.
+Labels: string, function-defn, int, string-template-expr, string-literal
+
+function errorFunction() {
+    string a1 = "a1";
+    int _ = string `${a1}`; // @error expected 'int' but found 'string'
+    int a2 = 12;
+    int _ = string `${a2}`; // @error expected 'int' but found 'string'
+    int a3 = 13;
+    int _ = string `${a2 + a3}`; // @error expected 'int' but found 'string'
+    int _ = <int> string `${a2 + a3}`; // @error 'string' cannot be cast to 'int'
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
```

Comment:
We can improve the description by mentioning the types.

Reply:
Updated the description. Please check.